jupyter_server/serverapp.py

```diff
@@ -271,30 +291,30 @@ def init_settings(self, jupyter_app, kernel_manager, contents_manager,
         settings.update(settings_overrides)
         return settings
 
-    def init_handlers(self, settings):
+    def init_handlers(self, default_services, settings):
         """Load the (URL pattern, handler) tuples for each component."""
-
         # Order matters. The first handler to match the URL will handle the request.
         handlers = []
         # load extra services specified by users before default handlers
         for service in settings['extra_services']:
             handlers.extend(load_handlers(service))
-        handlers.extend([(r"/login", settings['login_handler_class'])])
-        handlers.extend([(r"/logout", settings['logout_handler_class'])])
-        handlers.extend(load_handlers('jupyter_server.files.handlers'))
-        handlers.extend(load_handlers('jupyter_server.view.handlers'))
-        handlers.extend(load_handlers('jupyter_server.nbconvert.handlers'))
-        handlers.extend(load_handlers('jupyter_server.kernelspecs.handlers'))
-        handlers.extend(load_handlers('jupyter_server.edit.handlers'))
-        handlers.extend(load_handlers('jupyter_server.services.api.handlers'))
-        handlers.extend(load_handlers('jupyter_server.services.config.handlers'))
-        handlers.extend(load_handlers('jupyter_server.services.kernels.handlers'))
-        handlers.extend(load_handlers('jupyter_server.services.contents.handlers'))
-        handlers.extend(load_handlers('jupyter_server.services.sessions.handlers'))
-        handlers.extend(load_handlers('jupyter_server.services.nbconvert.handlers'))
-        handlers.extend(load_handlers('jupyter_server.services.kernelspecs.handlers'))
-        handlers.extend(load_handlers('jupyter_server.services.security.handlers'))
-        handlers.extend(load_handlers('jupyter_server.services.shutdown'))
+
+        # Add auth services.
+        if 'auth' in default_services:
+            default_services.remove('auth')
+            handlers.extend([(r"/login", settings['login_handler_class'])])
+            handlers.extend([(r"/logout", settings['logout_handler_class'])])
+
+        # Load default services. Raise exception if service not found in JUPYTER_SERVICES.      
+        for service in default_services:
+            if service in JUPYTER_SERVICES:
+                for loc in JUPYTER_SERVICES[service]:
+                    handlers.extend(load_handlers(loc))
+            else:
+                raise Exception("{} is not recognized as a default "
+                                "service of jupyter_server.".format(service))
```

Comment:
>I wonder if issuing a warning message here would be more friendly

Hm, what would be most useful for developers?

A bad service in this list is a mistake by the developer (since this isn't a configurable trait). Should a server ever run when there's a mistake in this list? A bad service is either a typo or misplaced (custom services belong in the `extra_services` list). 

I *think* I'd expect the server to fail fast? As a developer, I don't want my subclass to go out in the wild and neglect some service I expected it to have (like `auth`)? 

I've also updated the error message to point to  `extra_services`.